File: DesignPatternsExercises/java/SingletonDesignPattern/Exercise/FamilyMember.java
class FamilyMember {
    __________(4)__________ balance;
    String name;

    public FamilyMember(String name) {
        this.name = name;
        this.balance = __________(5)__________;
    }

    public void withdraw(int amount) {
        __________(6)__________.withdraw(amount);
    }

    public void deposit(int amount) {
        __________(7)__________.deposit(amount);
    }

    public int getBalance() {
        return __________(8)__________.getBalance();
    }
}

File: DesignPatternsExercises/ruby/singleton_design_pattern/solution/family_member.rb
require_relative 'balance'

class FamilyMember
  def initialize(name)
    @name = name
    # @balance = Balance.__________(7)__________
    @balance = Balance.instance
  end

  def withdraw(amount)
    # __________(8)__________(amount)
    @balance.withdraw(amount)
  end

  def deposit(amount)
    # __________(9)__________(amount)
    @balance.deposit(amount)
  end

  def balance
    # __________(10)__________
    @balance.balance
  end
end
